prime-router/src/test/kotlin/fhirengine/utils/HL7ReaderTests.kt

```diff
@@ -15,11 +14,12 @@ class HL7ReaderTests {
     @Test
     fun `test decoding of bad HL7 messages`() {
         val actionLogger = ActionLogger()
-        val hl7Reader = HL7Reader(actionLogger)
+        val hl7Reader = HL7Reader()
 
         // Empty data
         val badData1 = ""
-        hl7Reader.getMessages(badData1)
+        val (_, errors) = hl7Reader.getMessages(badData1)
+        actionLogger.error(errors)
         assertThat(actionLogger.hasErrors()).isTrue()
         actionLogger.logs.clear()
```

Comment:
An example of the potential simplification, this entire test can be done without accessing the errors indirectly through the `actionLogger` allowing a stronger guarantee that it's testing the correct thing and an easier to read test.